plan/optimizer.go

```diff
@@ -152,6 +152,9 @@ func physicalOptimize(logic LogicalPlan) (PhysicalPlan, error) {
 	if err != nil {
 		return nil, errors.Trace(err)
 	}
+	if t.invalid() {
```

Comment:
When will it happen? Any tests?

Reply:
It's hard to construct a test for this PR. This issue was met in our customer support, instead of panic due to "nil pointer dereference", we throw an error to avoid "lost connection" to tidb-server